Code/GraphMol/Depictor/EmbeddedFrag.cpp

```diff
@@ -982,62 +940,58 @@ void EmbeddedFrag::mergeWithCommon(EmbeddedFrag &embObj,
+      // side check for crowding and reflect
       reflectIfNecessaryDensity(embObj, commAtms[0], commAtms[1]);
     } else {
       // finally if we have more than two atoms in common - we will use the
-      // third
-      // atom to figure out if we need a reflection12
+      // third atom to figure out if we need a reflection12
       reflectIfNecessaryThirdPt(embObj, commAtms[0], commAtms[1], commAtms[2]);
     }
   }
 
   // finally merge the fragment by copying the non common atoms
-  const INT_EATOM_MAP &oatoms = embObj.GetEmbeddedAtoms();
-  INT_EATOM_MAP_CI ori;
+  const auto &oatoms = embObj.GetEmbeddedAtoms();
   // copy the eatoms in embObj to this fragment
-  for (ori = oatoms.begin(); ori != oatoms.end(); ori++) {
-    int aid = ori->first;
+  for (const auto &ori : oatoms) {
+    auto aid = ori.first;
     if (std::find(commAtms.begin(), commAtms.end(), aid) == commAtms.end()) {
-      d_eatoms[aid] = ori->second;
+      d_eatoms[aid] = ori.second;
       // also if any of these atoms have unattached neighbors add them to the
       // queue
-      if (ori->second.neighs.size() > 0) {
+      if (ori.second.neighs.size() > 0) {
         if (std::find(d_attachPts.begin(), d_attachPts.end(), aid) ==
             d_attachPts.end()) {
           d_attachPts.push_back(aid);
         }
       }
     } else {
-      if (ori->second.CisTransNbr >= 0) {
-        d_eatoms[aid].CisTransNbr = ori->second.CisTransNbr;
-        d_eatoms[aid].normal = ori->second.normal;
-        d_eatoms[aid].ccw = ori->second.ccw;
+      if (ori.second.CisTransNbr >= 0) {
+        d_eatoms[aid].CisTransNbr = ori.second.CisTransNbr;
+        d_eatoms[aid].normal = ori.second.normal;
+        d_eatoms[aid].ccw = ori.second.ccw;
       }
-      if (ori->second.angle > 0.0) {
-        d_eatoms[aid].angle = ori->second.angle;
-        d_eatoms[aid].nbr1 = ori->second.nbr1;
-        d_eatoms[aid].nbr2 = ori->second.nbr2;
+      if (ori.second.angle > 0.0) {
+        d_eatoms[aid].angle = ori.second.angle;
+
+        d_eatoms[aid].nbr1 = ori.second.nbr1;
+        d_eatoms[aid].nbr2 = ori.second.nbr2;
       }
     }
   }
 
   // remember to update the not yet done neighbor of nbrAid
-  RDKit::INT_VECT_CI cai;
-  for (cai = commAtms.begin(); cai != commAtms.end(); cai++) {
-    this->updateNewNeighs((*cai));  //, mol);
+  for (auto cai : commAtms) {
+    this->updateNewNeighs(cai);
   }
 }
 
-void EmbeddedFrag::mergeFragsWithComm(
-    std::list<EmbeddedFrag> &efrags) {  //, const RDKit::ROMol *mol) {
+void EmbeddedFrag::mergeFragsWithComm(std::list<EmbeddedFrag> &efrags) {
   PRECONDITION(dp_mol, "");
   // first merge any fragments what share atoms in common
-  std::list<EmbeddedFrag>::iterator efri, nfri = efrags.end();
+  auto nfri = efrags.end();
   while (1) {
     RDKit::INT_VECT commAtms;
-    for (efri = efrags.begin(); efri != efrags.end(); efri++) {
+    for (auto efri = efrags.begin(); efri != efrags.end(); ++efri) {
```

Comment:
any reason not to do
```
for( const auto &efri : efrags)
```

here?

Reply:
yeah, we need the iterator to set `nfri` a couple lines down